make.py

```diff
@@ -0,0 +1,252 @@
+"""
+Helper script to build langkit and its dependencies.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+from contextlib import contextmanager
+
+import argparse
+
+import os
+import shutil
+import subprocess
+import sys
+
+
+from langkit.utils import add_search_path
+
+
+parser = argparse.ArgumentParser(
+    description='Helper script to build langkit and its dependencies'
+)
+parser.add_argument('--build-mode', default='dev')
+
+subparsers = parser.add_subparsers()
+
+build_parser = subparsers.add_parser('build')
+build_parser.set_defaults(command='build')
+
+setenv_parser = subparsers.add_parser('setenv')
+setenv_parser.set_defaults(command='setenv')
+
+install_parser = subparsers.add_parser('install')
+install_parser.set_defaults(command='install')
+install_parser.add_argument('prefix', type=str,
+                            help='Where to install Langkit dependencies')
+install_parser.add_argument('--gnat-install-dir', type=str, default=None,
+                            help='Specify where GNAT is installed. If not'
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+install_parser.add_argument('--gnatcoll-bindings-install-dir', type=str,
+                            default=None,
+                            help='Specify where GNATCOLL bindings are '
+                                 'installed. If not provided, assume it is '
+                                 'installed where Langkit must be installed.')
+install_parser.add_argument('--libgpr-install-dir', type=str,
+                            default=None,
+                            help='Specify where libgpr is installed. If not '
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+
+
+ROOT_DIR = os.path.dirname(__file__)
+
+if ROOT_DIR != '':
+    os.chdir(ROOT_DIR)
+
+LIBPYTHONLANG_DIR = os.path.join('contrib', 'python')
+
+
+@contextmanager
+def in_dir(path):
+    """
+    Move the working directly to the given path for the span of this context
+    manager.
+
+    :type path: str
+    """
+    old_dir = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(old_dir)
+
+
+def pip_install(setup_dir):
+    """
+    Install the python package located in the given directory.
+
+    :type setup_dir: str
+    :raise CalledProcessError if installation terminates with non-zero status
```

Comment:
I don’t think this is a correct Sphinx annotation. May I suggest moving this to a regular English sentence after the description? This is the style we have been using in Langkit to describe exceptions.